src/ts/components/labels/MetadataLabel.ts

```diff
@@ -57,18 +62,33 @@ export class MetadataLabel extends Label<MetadataLabelConfig> {
         case MetadataLabelContent.Description:
           this.setText(uiconfig.metadata.description);
           break;
+        case MetadataLabelContent.AdMessage:
+          this.setText(this.adMessage);
+          break;
       }
     };
 
     let unload = () => {
       this.setText(null);
+      this.adMessage = '';
     };
 
     // Init label
     init();
     // Clear labels when source is unloaded
     player.on(player.exports.PlayerEvent.SourceUnloaded, unload);
 
+    player.on(player.exports.PlayerEvent.AdStarted, (event) => {
+      const ad = (event as AdEvent).ad;
+      if (!ad.isLinear) {
+        return;
+      }
+
+      const linearAd = ad as LinearAd;
+      this.adMessage = linearAd.uiConfig?.message ?? '';
+      init();
+    });
```

Comment:
Following the previous comment can we do the following here?

```suggestion
    player.on(player.exports.PlayerEvent.AdStarted, (event) => {
      if (config.context != MetadataLabelContent.AdMessage) { return }
      
      const ad = (event as AdEvent).ad;
      if (!ad.isLinear) {
        return;
      }

      const linearAd = ad as LinearAd;
      this.setText(linearAd.uiConfig?.message ?? '';);
    });
```

Reply:
Or: we could only subscribe the AdStarted even when the type is AdMessage. This was we don't need the early return at all.

Reply:
The new code has now been extracted into its own class, so this should not apply anymore from e5b32fc7e069a1dd4923f3ffa84203c550a1a7f2